### PR TITLE
[#135] Add cache consistency check

### LINKED ISSUE #135: Add a check for cache consistency
This sets a couple items from the web and from cron and checks how consistent they are and detect a split brain scenario

* [ ] This should actually check consistency of 3 places web vs cron vs webservice

### PR DESCRIPTION
**Pull request by matthewhilton**
Closes #135 , Based on Brendans PR https://github.com/catalyst/moodle-tool_heartbeat/pull/149

## Changes
- Adds cache check, which checks the get_config (cached) and DB values (not cached) in cron and web environments.
- If a difference is reported, it means there is a cache mismatch / config issue somewhere.
- (Unrelated to issue) Updated README and moved branches to top, and added php versions (these are the versions listed by Moodle docs for each version)

## Example output:
Screenshot 2023-10-24 at 08-35-29 mheartbeat-cache Administration Reports System status

### CODE REVIEW

**Comment by matthewhilton** [comment]
## TODO after merging:

- [ ] Cherry pick to MOODLE_39_STABLE branch

File: db/tasks.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @copyright  Catalyst IT 2023
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+$tasks = [
+    [
+        'classname' => 'tool_heartbeat\task\cachecheck',
+        'minute' => '*',
```
Comment (brendanheywood) [inline db/tasks.php:29]: give that ping only updates once a day, lets only run this on a similar cadence, maybe once an hour, or twice a day

Comment (matthewhilton) [reply db/tasks.php:29]: Good idea - the task is very lightweight so probably not necessary. I have changed it to every 8 hrs (arbitrary)

File: README.md [hunk 2]
```diff
@@ -20,6 +20,19 @@ NOTE: In an ideal world this plugin should be redundant and most of it's functio
 
 https://tracker.moodle.org/browse/MDL-47271
 
+# Branches
+
+| Branch             | Moodle version | PHP Version |
+| ------------------ | -------------- | ----------- |
+| master             | Moodle 2.7 +   | Php 5.4.4+  |
```
Comment (brendanheywood) [inline README.md:27]: missing the max version for master

Comment (matthewhilton) [reply README.md]: Good catch - have added

**Pull request merged**